crates/sui-core/src/consensus_adapter.rs

```diff
@@ -633,17 +656,22 @@ impl ConsensusAdapter {
         // this means we might be sending transactions from previous epochs to narwhal of
         // new epoch if we have not had this barrier.
         epoch_store
-            .within_alive_epoch(self.submit_and_wait_inner(transaction, &epoch_store))
+            .within_alive_epoch(self.submit_and_wait_inner(transactions, &epoch_store))
             .await
             .ok(); // result here indicates if epoch ended earlier, we don't care about it
     }
 
     #[allow(clippy::option_map_unit_fn)]
     async fn submit_and_wait_inner(
         self: Arc<Self>,
-        transaction: ConsensusTransaction,
+        transactions: Vec<ConsensusTransaction>,
         epoch_store: &Arc<AuthorityPerEpochStore>,
     ) {
+        assert!(transactions.len() > 0);
```

Comment:
This assertion is problematic. The current logic might end up with an empty cert list since it currently allow skipping already processed certificates.